pkg/querier/queryrange/roundtrip.go

```diff
@@ -255,6 +261,33 @@ func validateMaxEntriesLimits(req *http.Request, reqLimit uint32, limits Limits)
 	return nil
 }
 
+func validateMatchers(req *http.Request, limits Limits, matchers []*labels.Matcher) error {
+	tenants, err := tenant.TenantIDs(req.Context())
+	if err != nil {
+		return httpgrpc.Errorf(http.StatusBadRequest, err.Error())
```

Comment:
Why are we returning an httpgrpc error here? This is not a gRPC handler, so rather let the calling function decide how to return this error.